Core/include/AnalysisMath.h

```diff
@@ -109,6 +109,93 @@ double Calculate_visiblePzeta(const LVector1& l1_p4, const LVector2& l2_p4)
+
+    double d1 = pow(std::abs(a1.mass() - mass_top),2) + pow (std::abs(b1.mass() - mass_top),2);
+    double d2 = pow(std::abs(a2.mass() - mass_top),2) + pow (std::abs(b2.mass() - mass_top),2);
+    double d3 = pow(std::abs(a3.mass() - mass_top),2) + pow (std::abs(b3.mass() - mass_top),2);
+    double d4 = pow(std::abs(a4.mass() - mass_top),2) + pow (std::abs(b4.mass() - mass_top),2);
+    std::pair<double, double> pair_mass_top;
+    if (d1<d2 && d1<d3 && d1<d4) {
+        pair_mass_top.first = a1.mass();
+        pair_mass_top.second = b1.mass();
+    }
+    if (d2<d1 && d2<d3 && d2<d4) {
+        pair_mass_top.first = a2.mass();
+        pair_mass_top.second = b2.mass();
+    }
+    if (d3<d1 && d3<d2 && d3<d4) {
+        pair_mass_top.first = a3.mass();
+        pair_mass_top.second = b3.mass();
+    }
+    if (d4<d1 && d4<d3 && d4<d2) {
+        pair_mass_top.first = a4.mass();
+        pair_mass_top.second = b4.mass();
+    }
+    return pair_mass_top;
+}
+
+template<typename LVector1, typename LVector2, typename LVector3 >
+double Calculate_dR_boosted(const LVector1& particle_1, const LVector2& particle_2, const LVector3& h){
+    const analysis::LorentzVectorXYZ h_vector(h.px(),h.py(),h.pz(),h.e());
+    const auto boosted_1 = ROOT::Math::VectorUtil::boost(particle_1, h_vector.BoostToCM());
+    const auto boosted_2 = ROOT::Math::VectorUtil::boost(particle_2, h_vector.BoostToCM());
+    return ROOT::Math::VectorUtil::DeltaR(boosted_1, boosted_2); // R between the two final state particle in the h rest frame
+}
+
+
+template<typename LVector1, typename LVector2, typename LVector3, typename LVector4, typename LVector5,  typename LVector6>
+double Calculate_phi_4bodies(const LVector1& lepton1, const LVector2& lepton2, const LVector3& bjet1, const LVector4& bjet2, const LVector5& svfit, const LVector6& bb){
+    const auto H = bb + svfit;
+    const analysis::LorentzVectorXYZ vec_H(H.px(), H.py(),H.pz(),H.e());
+    const auto boosted_l1 = ROOT::Math::VectorUtil::boost(lepton1, vec_H.BoostToCM());
+    const auto boosted_l2 = ROOT::Math::VectorUtil::boost(lepton2, vec_H.BoostToCM());
+    const auto boosted_j1 = ROOT::Math::VectorUtil::boost(bjet1, vec_H.BoostToCM());
+    const auto boosted_j2 = ROOT::Math::VectorUtil::boost(bjet2, vec_H.BoostToCM());
+    const TVector3 vec_l1(boosted_l1.px(),boosted_l1.py(),boosted_l1.pz());
+    const TVector3 vec_l2(boosted_l2.px(),boosted_l2.py(),boosted_l2.pz());
+    const TVector3 vec_j1(boosted_j1.px(),boosted_j1.py(),boosted_j1.pz());
+    const TVector3 vec_j2(boosted_j2.px(),boosted_j2.py(),boosted_j2.pz());
+    const auto n1 = vec_l1.Cross(vec_l2);
+    const auto n2 = vec_j1.Cross(vec_j2);
+    return ROOT::Math::VectorUtil::Angle(n1, n2); //angle between the decay planes of the four final state elements expressed in the H rest frame
+}
+
+template<typename LVector1, typename LVector2>
+double Calculate_theta_star(const LVector1& svfit, const LVector2& bb){
```

Comment:
It is better to return cos(theta_star), so function will be called Calculate_cosThetaStar
better to call first parameter ll instead of svfit